static/coffee/controllers.coffee

```diff
@@ -235,6 +240,55 @@ controllers.controller('MessagesController', ['$scope', '$timeout', '$interval',
       $scope.setAdvancedSearch(false)
     )
 
+  $scope.poll = ->
+    # a poll is already in progress, skip this one
+    if $scope.pollBusy
+      return
+
+    $scope.pollBusy = true
+    $scope.activeSearchRefresh = $scope.buildSearch()
+    $scope.activeSearchRefresh.last_refresh = $scope.lastPollTime
+    $scope.activeSearchRefresh.after = $scope.lastPollTime
+
+    MessageService.fetchOld($scope.activeSearchRefresh, $scope.lastPollTime, $scope.oldItemsPage).then((data) ->
+      $scope.lastPollTime = new Date()
+      $scope.pollBusy = false
+
+      # quick access to index of items
+      scopeItems = {}
+      for item, i in $scope.items
+        scopeItems[item.id] = i
+
+      for item in data.results
+        if scopeItems.hasOwnProperty(item.id)
+          # the item exists so replace with new data
+          $scope.items[scopeItems[item.id]] = item
+          $scope.updateItems()
+        else if !item.busy
+          # new item so we add it to the top
+          $scope.items.unshift(item)
```

Comment:
Am I understanding this correctly: items that are new but are busy should not be rendered?